core/translate/expr.rs

```diff
@@ -698,7 +698,93 @@ pub fn translate_expr(
             }
             Ok(target_register)
         }
-        ast::Expr::Case { .. } => todo!(),
+        ast::Expr::Case {
+            base,
+            when_then_pairs,
+            else_expr,
+        } => {
+            // There's two forms of CASE, one which does quality against the when values:
+            //   CASE [expr] WHEN [value] ELSE [value] END
+            // And one which evaluates a series of boolean predicates:
+            //   CASE (WHEN [bool_expr] THEN [value])+ END
+            // This just changes which sort of branching instruction to issue, after we
+            // generate the expression if needed.
+            let return_label = program.allocate_label();
+            let mut next_case_label = program.allocate_label();
+            let base_reg = base.as_ref().map(|_| program.alloc_register());
+            let expr_reg = program.alloc_register();
+            if let Some(base_expr) = base {
+                translate_expr(
+                    program,
+                    referenced_tables,
+                    base_expr,
+                    base_reg.unwrap(),
+                    precomputed_exprs_to_registers,
+                )?;
+            };
+            for (when_expr, then_expr) in when_then_pairs {
+                translate_expr(
+                    program,
+                    referenced_tables,
+                    when_expr,
+                    expr_reg,
+                    precomputed_exprs_to_registers,
+                )?;
+                match base_reg {
+                    Some(base_reg) => program.emit_insn_with_label_dependency(
+                        Insn::Ne {
+                            lhs: base_reg,
+                            rhs: expr_reg,
+                            target_pc: next_case_label,
+                        },
+                        next_case_label,
+                    ),
+                    None => program.emit_insn_with_label_dependency(
+                        Insn::IfNot {
+                            reg: expr_reg,
+                            target_pc: next_case_label,
+                            null_reg: 1,
```

Comment:
I, again, do not understand what this argument is about :shrug:

Reply:
https://github.com/sqlite/sqlite/blob/c29bc733beaff046e231db7681197df726657acc/src/expr.c#L6145

Clearly it's some kind of flag that is conditionally set when sqlite wants to make the jump not only in the case where reg is 0, but also when it is NULL. There's also a SQLITE_JUMPIFNULL. Still not sure when exactly this option is set. In any case, we should probably rename the field to `jump_if_null` instead of `null_reg`